mmf/trainers/core/training_loop.py

```diff
@@ -56,37 +56,19 @@ def run_training_epoch(self) -> None:
             # Seed the sampler in case if it is distributed
             self.dataset_loader.seed_sampler("train", self.current_epoch)
 
-            # For iterable datasets we cannot determine length of dataset properly.
-            # For those cases we set num_remaining_batches to be the (number of
-            # updates remaining x update_frequency)
-            num_remaining_batches = (
-                (
-                    (self.max_updates - self.num_updates)
-                    * self.training_config.update_frequency
-                )
-                if isinstance(
-                    self.train_loader.current_dataset, torch.utils.data.IterableDataset
-                )
-                else len(self.train_loader)
-            )
-
-            combined_report = None
-            num_batches_for_this_update = 1
+            should_start_update = True
             for idx, batch in enumerate(self.train_loader):
-
-                if (idx + 1) % self.training_config.update_frequency == 0:
+                if should_start_update:
                     combined_report = None
-                    num_batches_for_this_update = min(
-                        self.training_config.update_frequency, num_remaining_batches
-                    )
-
```

Comment:
Is the case for remaining batches < self.training_config.update_frequency handled?

Reply:
@vedanuj, very good point - I think here the default behavior should be not doing the update if the number of iterations in the current update has not reached update_frequency yet -- bc users of this framework probably will not expect the last batch to have a high variance in their loss values.

What I mean is this: If the update_frequency is 100, and the number of batches we currently have done is 210, this means that we have already done 2 updates (2 * 100). And since 10 < 100, we will not be doing another update. Because the loss accumulated from the 10 iterations could be of high variance, which is less desirable to training especially having it be the last update. 

Let me know if you think otherwise. :)

Reply:
I looked into pytorch lightning's [implementation](https://github.com/PyTorchLightning/pytorch-lightning/blob/e1c152b13c60f064a312f97ad0e326319bfda047/pytorch_lightning/trainer/training_loop.py#L853). It seems like they do perform a final update. Will update this PR.